persistence_legacy.go

```diff
@@ -271,3 +320,61 @@ func (db sqlitePersistence) UpdateMessageOutgoingStatus(id string, newOutgoingSt
 	`, newOutgoingStatus, id)
 	return err
 }
+
+// BlockContact updates a contact, deletes all the messages and 1-to-1 chat, updates the unread messages count and returns a map with the new count
+func (db sqlitePersistence) BlockContact(contact Contact) ([]*Chat, error) {
+	var (
+		tx  *sql.Tx
+		err error
+	)
+	tx, err = db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
```

Comment:
Maybe we could write a simple helper function that commits or rollbacks a tx based on received err, tx arguments?

Reply:
I might do a separate PR refactoring once the code settles a bit, but good idea